src/database/sql_database.cc

```diff
@@ -1193,10 +1192,27 @@ int SQLDatabase::createContainer(int parentID, std::string name, const std::stri
 
 int SQLDatabase::insert(const char* tableName, const std::vector<SQLIdentifier>& fields, const std::vector<std::string>& values, bool getLastInsertId)
 {
+    assert(fields.size() == values.size());
     auto sql = fmt::format("INSERT INTO {} ({}) VALUES ({})", identifier(tableName), fmt::join(fields, ","), fmt::join(values, ","));
     return exec(sql, getLastInsertId);
 }
 
+void SQLDatabase::insertMultipleRows(const char* tableName, const std::vector<SQLIdentifier>& fields, const std::vector<std::vector<std::string>>& valuesets)
```

Comment:
These const ref parameters look like they can be replaced by &&

Reply:
I tried to keep it similar to insert(), and since there is no transfer of ownership intended, I see no point in doing so.

Reply:
The point is to avoid copies.

Reply:
Just tested. Small size reduction by doing so.

Reply:
What do you mean exactly? The fields/valuesets references?

Reply:
Change const std::vector& to std::vector&& and add std::move. Done here: https://github.com/neheb/gerbera/commit/b7a721698a5545379f9c6160dab7695ca8c913b4

Reply:
If you really insist on that change, I can do it. But I'm not a friend of this way. There should not be any copy when passing a reference and when adding std::vector&&, there is a high risk that in future code, someone forgets the std::move and then there will be a copy. Also, I should change the parameters of insert(), too - to be consistent.
What (code?) size reduction are we talking about and is it significant?

Reply:
How? && forces std::move.

Size reduction is there but not significant.

Reply:
> && forces std::move

True, this will force a move.

Feel free to change the interface and submit a PR. But similar to insert(), if I wanted to call the function twice with the same vector and for example reuse fields, I must assume that insertMultipleRows moves away the data (even though it will not move it anywhere). I'm OK with a small increase in size (what feature doesn't) and preserve code readabiliy and ease of understanding without the need for extra documentation.